### PR TITLE
docs(VDataTable): add missing props to group.summary slot

### PR DESCRIPTION
**Pull request by Daanra**
## Description

Add the missing isOpen and toggle props to the group.summary slot of the VDataTable component.

## Motivation and Context

The group.summary slot doesn't have the `isOpen` and `toggle` props. The `isOpen` prop would be useful for hiding (parts of) the summary when the group is closed. The `toggle` prop might also be useful if a developer wants the user to be able to close the group from within the summary.
## How Has This Been Tested?

Visually (playground)

## Markup:

<details>

```vue

<template>
  <v-container>
    <v-data-table :items="items" :headers="headers" group-by="type">
      <template #group.summary="{ isOpen, toggle }">
        <v-btn icon @click="toggle">
          <v-icon>
            {{ isOpen ? '$minus' : '$plus' }}
          </v-icon>
        </v-btn>
      </template>
    </v-data-table>
  </v-container>
</template>

<script>
  export default {
    data: () => ({
      items: [{
        type: 1,
        name: 'First',
      }, {
        type: 2,
        name: 'Second',
      }, {
        type: 1,
        name: 'Third'
      }],
      headers: [{
        text: 'Type',
        value: 'type'
      }, {
        text: 'Name',
        value: 'name',
      }]
    })
  }
</script>

```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VDataTable/VDataTable.ts [hunk 1]
```diff
@@ -355,7 +355,14 @@ export default VDataIterator.extend({
 
       if (this.$scopedSlots['group.summary']) {
         children.push(this.$createElement('template', { slot: 'column.summary' }, [
-          this.$scopedSlots['group.summary']!({ group, groupBy: props.options.groupBy, items, headers: this.computedHeaders }),
+          this.$scopedSlots['group.summary']!({
```
Comment (johnleider) [inline packages/vuetify/src/components/VDataTable/VDataTable.ts:358]: While I agree with you in terms of code styling, that is out of scope of this PR. You are welcome to make that update in a different PR.

Comment (Daanra) [reply packages/vuetify/src/components/VDataTable/VDataTable.ts]: Changed it back to a single line
